lib/filewatch/discover.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'watched_file'
+
+module FileWatch
+  class Discover
+    attr_reader  :logger, :files, :wf_vars
+
+    def initialize(opts, loggr)
+      @logger = loggr
+      @watching = []
+      @exclude = []
+      @files = {}
+      set_ignore_older(opts[:ignore_older])
```

Comment:
may I suggest making these set_* functions pure?

`@ignore_older = parse_ignore_older(opts[:ignore_older])`

makes it less magical when the vars are used in the following line